### ISSUE TITLE
Selectable text outside of inputs/textareas

### ISSUE THREAD

**Issue by v-atamanenko** [mikke89/RmlUi#661]
Hello, I was wondering if RmlUi supports making elements mouse-selectable besides the standard inputs? As in, in CSS terms, `user-select: text` behavior as opposed to `user-select: none` as it is for everything now. And if doesn't, how complicated do you think is it to implement that? If it isn't something requiring major changes, I'd gladly contribute. Thank you.

**Comment by mikke89** [comment]
Hello!

You're right, we don't have a way to select elements or text outside of input fields. Out of curiosity, can I ask what you're use case is? I haven't really heard requests for this feature so far, so I am just curious.

I don't think it's a very simple feature to implement, but certainly possible. If you don't mind a challenge, I would encourage you to give it a go! Right now, all our selection logic is located in the text input widget, so you could start taking a look at that. The next step would be to apply some of this logic directly to the ElementText objects, which holds the strings and handles the text rendering. Then, it would be a matter of keeping track of the first and last elements that the user selects, and marking all elements and text between them as selected somehow.

**Comment by ShawnCZek** [comment]
Hey, I know some of our players have requested a feature to copy messages from the chat. However, as it is quite a rare functionality (among popular games, I know only of Counter-Strike 2 having this), we have never explored this option, especially as we have never needed this anywhere else.

**Comment by mikke89** [comment]
Right, I see how that could be useful, although I would put that in the relatively rare, non-crucial bucket. But it could be a nice-to-have feature if it isn't too invasive or end up being very complicated.

**Comment by v-atamanenko** [comment]
@mikke89 I don't know if you're familiar, but basically with every video game console SDK, vendors provide you sort of an all-in-one companion app to manage your devkit. In the PSVita homebrew community we use the community-made free/open-source SDK and one big thing it still lacks is such a companion app. So yeah, I'm working on it using RmlUi and here's how it looks so far:
![image](https://github.com/user-attachments/assets/a5471d17-db3d-46b1-9029-ab8ac0c90245)
I would love to use the feature in question for the logging output window to allow selecting and copying parts of it. Textarea isn't a good fit here because I also want to have optional extra columns like the timestamp.

Thanks for your detailed answer, I will look into it.

**Comment by mikke89** [comment]
Yeah, that does sounds quite reasonable, and I think you have convinced me that this is a feature that would be nice to have. I don't want to make any promises though until I know we can find an implementation without significant compromises. I'll happily support you and see where this goes, let me know if you need any help or pointers along the way!

I like the style by the way, has a nice retro look to it. Sounds like a very interesting project.
